src/syscheckd/CMakeLists.txt

```diff
@@ -87,9 +89,10 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
     target_compile_definitions(wazuh-syscheckd-event PUBLIC -D_WIN32_WINNT=0x600 -DEVENTCHANNEL_SUPPORT)
 else()
     add_executable(wazuh-syscheckd ${SYSCHECKD_SRC})
+    set_target_properties(wazuh-syscheckd PROPERTIES LINKER_LANGUAGE CXX)
```

Comment:
Is this necessary?

Reply:
Yes, this was necessary to force the use of the C++ linker.
Without it, we encountered unresolved symbol errors for basic C++ functions, such as those in std::basic_string.